hugr-cli/src/main.rs

```diff
@@ -1,22 +1,32 @@
 //! Validate serialized HUGR on the command line
 
+use clap::Parser as _;
 use hugr_core::std_extensions::arithmetic::{
     conversions::EXTENSION as CONVERSIONS_EXTENSION, float_ops::EXTENSION as FLOAT_OPS_EXTENSION,
     float_types::EXTENSION as FLOAT_TYPES_EXTENSION, int_ops::EXTENSION as INT_OPS_EXTENSION,
     int_types::EXTENSION as INT_TYPES_EXTENSION,
 };
 use hugr_core::std_extensions::logic::EXTENSION as LOGICS_EXTENSION;
 
+use hugr_cli::{validate, CliArgs};
 use hugr_core::extension::{ExtensionRegistry, PRELUDE};
 
-use hugr_cli::CmdLineArgs;
-
-use clap::Parser;
 use clap_verbosity_flag::Level;
 
 fn main() {
-    let opts = CmdLineArgs::parse();
+    match CliArgs::parse() {
+        CliArgs::Validate(args) => run_validate(args),
+        CliArgs::External(_) => {
+            // TODO: Implement support for external commands.
+            // Running `hugr COMMAND` would look for `hugr-COMMAND` in the path
+            // and run it.
+            eprintln!("External commands are not supported yet.");
```

Comment:
This must be implemented before the next release, as we have dependents that use the library to define cli tools.

Reply:
Sure. This PR is mostly to make space so that other builtin CLI commands can be added. This is also why this does not close https://github.com/CQCL/hugr/issues/1207, which consists of two parts:
1. Introduce subcommands.
2. Add support for external subcommands.